drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -521,6 +517,35 @@ static int mtd_spi_nor_write(mtd_dev_t *mtd, const void *src, uint32_t addr, uin
     return 0;
 }
 
+static int mtd_spi_nor_write_page(mtd_dev_t *mtd, const void *src, uint32_t page, uint32_t offset,
+                                  uint32_t size)
+{
+    const mtd_spi_nor_t *dev = (mtd_spi_nor_t *)mtd;
+
+    DEBUG("mtd_spi_nor_write_page: %p, %p, 0x%" PRIx32 ", 0x%" PRIx32 ", 0x%" PRIx32 "\n",
+          (void *)mtd, src, page, offset, size);
+
+    size_t remaining = mtd->page_size - offset;
+    size = MIN(remaining, size);
```

Comment:
This compares a `size_t` with an `uint32_t` and is casted later to an `int`. I think this is asking for truncation issues :fearful:

Reply:
I can change it to use `uint32_t`.
The return value is <= page size, so it should fit into an `int`.

Reply:
I can't find an example where the page size is more than 32KB on a non internal flash device, so I agree. How about a line or two in the mtd.h file to explain that an `int` is safe to use, and add an assert in the mtd implementation to assert that claim?

Reply:
How about 

```C
    /* Implementation assumes page size is <= INT_MAX and a power of two. */
    /* We didn't find hardware yet where this is not true.                */
    assert(mtd->page_size <= INT_MAX);
    assert(__builtin_popcount(mtd->page_size));
```

Reply:
Then why not use `bitarithm_bits_set`? :)